pandas/io/pytables.py

```diff
@@ -4405,6 +4406,9 @@ def _unconvert_string_array(data, nan_rep=None, encoding=None):
                 dtype = "U{0}".format(itemsize)
             else:
                 dtype = "S{0}".format(itemsize)
+            # fix? issue #10366
+            data = _convert_string_array(data, _ensure_encoding(encoding),
```

Comment:
this is not needed
an exception is thrown on some conversions already this is duplicative